src/librbd/internal.cc

```diff
@@ -1941,6 +1942,126 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
     return r;
   }
 
+  int copy_deep(ImageCtx *src, IoCtx& dest_md_ctx, const char *destname,
```

Comment:
Nit: can we add this API method to ```librbd::api::Image``` instead of internal.cc? I am trying to kill this giant file off.